pkg/operator/sync.go

```diff
@@ -90,3 +86,124 @@ func (optr *Operator) waitForDeploymentRollout(resource *appsv1.Deployment) erro
 		return false, nil
 	})
 }
+
+func newDeployment(config OperatorConfig) *appsv1.Deployment {
+	replicas := int32(1)
+	template := newPodTemplateSpec(config)
+
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "machine-api-controllers",
+			Namespace: config.TargetNamespace,
+			Labels: map[string]string{
+				"api":     "clusterapi",
+				"k8s-app": "controller",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"api":     "clusterapi",
+					"k8s-app": "controller",
+				},
+			},
+			Template: *template,
+		},
+	}
+}
+
+func newPodTemplateSpec(config OperatorConfig) *corev1.PodTemplateSpec {
+	containers := newContainers(config)
+	tolerations := []corev1.Toleration{
+		{
+			Key:    "node-role.kubernetes.io/master",
+			Effect: corev1.TaintEffectNoSchedule,
+		},
+		{
+			Key:      "CriticalAddonsOnly",
+			Operator: corev1.TolerationOpExists,
+		},
+		{
+			Key:      "node.alpha.kubernetes.io/notReady",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+		{
+			Key:      "node.alpha.kubernetes.io/unreachable",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+	}
+
+	_true := true
```

Comment:
should be this call `runAsNonRoot`

Reply:
check https://github.com/openshift/machine-api-operator/pull/280/files/f9c605b0ee5d50bb86e2ac725738cdf7fc4c6c1f..c92f7f847848375db341a76ed4f56f9c35cbeea1#diff-fa45321336db7ad1cedc28bf643a4f97R153

Reply:
broken link

Reply:
sorry for that, I do not want to pass again the link, because it will be broken on the next push, but you can search for :
```
			SecurityContext: &corev1.PodSecurityContext{
				RunAsNonRoot: pointer.BoolPtr(true),
				RunAsUser:    &user,
			},
```